src/unix/thread.c

```diff
@@ -500,7 +504,102 @@ int uv_sem_trywait(uv_sem_t* sem) {
   return 0;
 }
 
-#else /* !(defined(__APPLE__) && defined(__MACH__)) */
+#elif defined(__GLIBC__) && __GLIBC_MINOR__ < 21
```

Comment:
This needs to be either unconditional on the version or detect the version at runtime (could use `gnu_get_libc_version()` for that) because the build time and runtime versions can be different.

Reply:
@bnoordhuis I’ve removed the version check for now.

How would checking at runtime work best? I guess we don’t want to parse the version every time one of these functions is called, so maybe using some `__attribute((constructor))` function that sets some local variable which keeps track of whether we need this is a good idea?

Reply:
Yep, that could work, or a `uv_once_t` in `uv_sem_init()`.

Reply:
@bnoordhuis I went with that … could you take another look at the most recent commit?